File: app/src/main/java/com/shicheeng/copymanga/json/MainBannerJson.java
package com.shicheeng.copymanga.json;

import android.util.Log;

import com.google.gson.JsonArray;
import com.google.gson.JsonElement;
import com.google.gson.JsonObject;
import com.google.gson.JsonParser;
import com.shicheeng.copymanga.KeyWordSwap;
import com.shicheeng.copymanga.apiName;
import com.shicheeng.copymanga.data.BannerList;

import java.io.IOException;
import java.util.ArrayList;
import java.util.HashMap;
import java.util.List;

import okhttp3.OkHttpClient;
import okhttp3.Request;
import okhttp3.Response;

public class MainBannerJson {

    public static JsonObject getMianList() throws IOException {
        OkHttpClient client = new OkHttpClient();
        Request request = new Request.Builder().url(apiName.mangaMainPage)
                .removeHeader(KeyWordSwap.USER_AGENT_WORD)
                .addHeader(KeyWordSwap.USER_AGENT_WORD, KeyWordSwap.FAKE_USER_AGENT)
                .build();

        try (Response response = client.newCall(request).execute()) {
            assert response.body() != null;
            String jsonData = response.body().string();

            return JsonParser.parseString(jsonData).getAsJsonObject().getAsJsonObject("results");
        }

    }

    public static List<BannerList> getBannerMain(JsonObject jsonObject) {

        JsonArray array1 = jsonObject.get("banners").getAsJsonArray();
        List<BannerList> bannerLists = new ArrayList<>();
        for (int i = 0; i < array1.size(); i++) {
            JsonElement element = array1.get(i).getAsJsonObject().get("type");
            if (element.getAsInt() == 1) {
                BannerList list = new BannerList();
                list.setJsonObject(array1.get(i).getAsJsonObject());
                bannerLists.add(list);
            }
        }
        return bannerLists;
    }

    public static JsonArray getRecMain(JsonObject jsonObject) {
        return jsonObject.get("recComics").getAsJsonObject().getAsJsonArray("list");

    }

    public static JsonArray getHotMain(JsonObject jsonObject) {
        return jsonObject.get("hotComics").getAsJsonArray();
    }

    public static JsonArray getNewMain(JsonObject jsonObject) {
        return jsonObject.get("newComics").getAsJsonArray();
    }

    public static JsonArray getFinishMain(JsonObject object) {
        return object.getAsJsonObject("finishComics").getAsJsonArray("list");
    }

    public static HashMap<Integer, JsonArray> getDayRankMain(JsonObject jsonObject) {
        HashMap<Integer, JsonArray> map = new HashMap<>();
        map.put(0, jsonObject.get("rankDayComics").getAsJsonObject().getAsJsonArray("list"));
        map.put(1, jsonObject.get("rankWeekComics").getAsJsonObject().getAsJsonArray("list"));
        map.put(2, jsonObject.get("rankMonthComics").getAsJsonObject().getAsJsonArray("list"));

        return map;

    }

}

File: app/src/main/java/com/shicheeng/copymanga/json/MainBannerJson.kt
package com.shicheeng.copymanga.json

import android.content.SharedPreferences
import com.google.gson.JsonArray
import com.google.gson.JsonObject
import com.shicheeng.copymanga.data.BannerList
import com.shicheeng.copymanga.util.parserToJson
import kotlinx.coroutines.Dispatchers
import kotlinx.coroutines.withContext
import okhttp3.OkHttpClient
import okhttp3.Request
import javax.inject.Inject
import javax.inject.Singleton

@Singleton
class MainBannerJson @Inject constructor(
    private val sharedPreferences: SharedPreferences,
) {

    private val apiHeader
        get() = sharedPreferences.getString("key_api_header_select", "copymanga.net")
            ?: "copymanga.net"
    private val mainPageUrl =
        "https://api.$apiHeader/api/v3/h5/homeIndex?platform=3&amp;format=json"

    @Inject
    lateinit var okHttpClient: OkHttpClient

    suspend fun fetchMainListData(): JsonObject = withContext(Dispatchers.Default) {
        val request: Request = Request.Builder()
            .url(mainPageUrl)
            .build()
        okHttpClient.newCall(request).execute().use { response ->
            val jsonData = requireNotNull(response.body).string()
            jsonData.parserToJson().asJsonObject.getAsJsonObject("results")
        }
    }


    fun getBannerMain(jsonObject: JsonObject): ArrayList<BannerList> {
        val array1 = jsonObject["banners"].asJsonArray
        val bannerLists = ArrayList<BannerList>()
        for (ele in array1) {
            val element = ele.asJsonObject["type"]
            if (element.asInt == 1) {
                val list = BannerList()
                list.jsonObject = ele.asJsonObject
                bannerLists.add(list)
            }
        }
        return bannerLists
    }

    fun getRecMain(jsonObject: JsonObject): JsonArray =
        jsonObject["recComics"].asJsonObject.getAsJsonArray("list")

    fun getHotMain(jsonObject: JsonObject): JsonArray = jsonObject["hotComics"].asJsonArray

    fun getNewMain(jsonObject: JsonObject): JsonArray = jsonObject["newComics"].asJsonArray

    fun getFinishMain(jsonObject: JsonObject): JsonArray {
        return jsonObject.getAsJsonObject("finishComics").getAsJsonArray("list")
    }

    fun getRecTopic(jsonObject: JsonObject): JsonArray {
        return jsonObject["topics"].asJsonObject["list"].asJsonArray
    }

    fun getDayRankMain(jsonObject: JsonObject): HashMap<Int, JsonArray> {
        val map = HashMap<Int, JsonArray>()
        map[0] = jsonObject["rankDayComics"].asJsonObject.getAsJsonArray("list")
        map[1] = jsonObject["rankWeekComics"].asJsonObject.getAsJsonArray("list")
        map[2] = jsonObject["rankMonthComics"].asJsonObject.getAsJsonArray("list")
        return map
    }
}